server/apps/main/tests/test_helpers.py

```diff
@@ -54,4 +55,44 @@ def test_moderator_check(self):
         Also ensure that it works for users in >1 group
         """
         self.assertFalse(is_moderator(self.non_moderator_user.user)) # non-moderator should be false
-        self.assertTrue(is_moderator(self.moderator_user.user)) # moderator user should be true
\ No newline at end of file
+        self.assertTrue(is_moderator(self.moderator_user.user)) # moderator user should be true
+
+    @vcr.use_cassette('server/apps/main/tests/fixtures/my_stories.yaml',
+                      record_mode='none', filter_query_parameters=['access_token'])
+    def test_fetch_by_uuid_sucesss(self):
+        """
+        Test that fetching an OH file by UUID works. 
+        Should succeed if no duplicate UUID in account
+        """
+        target_uuid = "ebad2890-bc43-11ed-95a4-0242ac130003"
+        response = get_oh_file(self.moderator_user, target_uuid)
+        self.assertEqual(response['id'], 63199447)
+
+
+    @vcr.use_cassette('server/apps/main/tests/fixtures/my_stories_duplicate_uuid.yaml',
+                      record_mode='none', filter_query_parameters=['access_token'])
+    def test_fetch_by_uuid_failure(self):
+        """
+        Test that fetching an OH file by UUID works. 
+        Should faile due to duplicate UUID in account
+        """
+        target_uuid = "ebad2890-bc43-11ed-95a4-0242ac130003"
+        with self.assertRaises(Exception): 
+            get_oh_file(self.moderator_user, target_uuid)
+
+
+    def test_make_tags(self):
+
+        # Create mock set of tags 
+        # Mock tags include 3 true items from the list, 2 false items from the list
+        tag_data = {'abuse': True, 'violence': True,
+                    'drug': False, 'mentalhealth': False,'negbody': True,
+                    'other': '',
+                    'moderation_status': 'approved',
+                    'moderation_comments': ''
+                    }
+        returned_data = make_tags(tag_data)
+        print(returned_data)
```

Comment:
After writing this test, I think we actually might want to modify the `make_tags` function. Currently the call as done here returns the following: 

```
['abuse', 'violence', '', '', 'negative body', None]
```

The `None` is because the `moderation_status` response is not reflected in the key at all, the empty strings are due to how `make_tags` returns empty strings. I think more optimally it should only return the 3 actually present tags and nothing else? 

@helendduncan What do you think?

Reply:
I don't think this function is used now - looking at it

Reply:
I think it's still used in the `upload` helper function which is called in the share experience-view (see https://github.com/alan-turing-institute/AutSPACEs/blob/c1796a78309f238f7ea963135032f00dbcba8140/server/apps/main/views.py#L122)

Reply:
So in response to the original comment:

Reply:
The use for the tags was to easily access all the information for the "my stories" page.The moderation status tag is used on a "per-story" basis still so given our current set up it would be good to have that stored in the metadata.

Reply:
> I think to me in the example above we would like to switch the list of triggers to be `['abuse', 'violence', 'negative body']`, so I think this would be _keep only the trigger warning tags_, as those trigger labels that don't apply won't show up. Does that make sense? :)

Typing at the same time! Yes that does make sense and I agree. I think we do still need to preserve the other information in the metadata but they needn't be together.

Reply:
Yep, perfect! 

I guess the moderation status is stored in the `PublicExperience` object in our AutSPACEs database and additionally it's also explicitly stored in the top-level metadata object as `"moderation_status":"not reviewed"` and similarly the trigger labels are also stored in the top-level item. It just seems that we also copy much of that information into the `tags` list. :)

Reply:
I've made that change with the last commit, it now only returns the non-empty tags and the test accounts for that.